docs/lang/articles/reference.md

```diff
@@ -255,3 +255,98 @@ with `expression2` as the error message.
 ### The `while` statement
 
 ### The `for` statement
+
+The `for` statement in Taichi is used to iterate over a range of numbers, multidimensional ranges, or the indices of elements in a field.
+
+```
+for_stmt ::=  "for" target_list "in" expression ":" suite
+```
+
+Taichi does not support `else` clause in `for` statements.
+
+The `for` loops can iterate in parallel if they are in the outermost scope.
+When a `for` loop is parallelized, the order of iteration is not determined,
+and it cannot be terminated by `break` statements.
+
+Taichi uses `ti.loop_config` function to set directives for the loop right after it.
+You can write `ti.loop_config(serialize=True)` before a range/ndrange `for` loop to let it run serially,
+then it can be terminated by `break` statements.
```

Comment:
Out of curiosity, when a loop is *not* serialized, do we have any AST checking mechanism to detect the invalid `break`?

If so, the AST transforming stage will also need to understand `loop_config`.

Reply:
Yes, the code is around https://github.com/taichi-dev/taichi/blob/master/taichi/ir/frontend_ir.cpp#L897